### ISSUE TITLE
Ajax

### ISSUE THREAD

**Issue by anishcana** [anishcana/jovial#8]
// Assume this is the data received from the backend
var responseData = [
    { name: "File 1", directory: "/path/to/directory1" },
    { name: "File 2", directory: "/path/to/directory2" },
    // Add more data as needed
];

// Function to populate the table with data
function populateTable(data) {
    var tbody = $('#dataTable tbody');
    tbody.empty(); // Clear existing table body content

    // Loop through the data and create table rows
    $.each(data, function(index, item) {
        var row = $('<tr>');
        row.append($('<td>').text(item.name));

        // Create a hyperlink for the directory column
        var directoryLink = $('<a>').attr('href', 'file://' + item.directory).text(item.directory);
        var directoryCell = $('<td>').append(directoryLink);
        row.append(directoryCell);

        tbody.append(row);
    });
}

// Call the function to populate the table with the received data
populateTable(responseData);

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html>
<head>
    <title>Upload Excel File</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script>
        $(document).ready(function() {
            $("#uploadForm").submit(function(event) {
                event.preventDefault();
                var formData = new FormData($(this)[0]);
                $.ajax({
                    url: "/upload",
                    type: "POST",
                    data: formData,
                    contentType: false,
                    processData: false,
                    success: function(response) {
                        // Handle success
                        console.log("File uploaded successfully:", response);
                    },
                    error: function(xhr, status, error) {
                        // Handle error
                        console.error("Error uploading file:", error);
                    }
                });
            });
        });
    </script>
</head>
<body>
    <h1>Upload Excel File</h1>
    <form id="uploadForm" enctype="multipart/form-data">
        <input type="file" name="file">
        <button type="submit">Upload</button>
    </form>
</body>
</html>

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html>
<head>
    <title>Upload Excel File</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="upload.js"></script>
</head>
<body>
    <h1>Upload Excel File</h1>
    <form id="uploadForm" enctype="multipart/form-data">
        <input type="file" name="file">
        <button type="submit">Upload</button>
    </form>
</body>
</html>

**Comment by anishcana** [comment]
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

@RestController
public class FileUploadController {

    @PostMapping("/upload")
    public String uploadFile(@RequestParam("file") MultipartFile file) {
        if (file.isEmpty()) {
            return "Please select a file to upload.";
        }
        // Process the file (e.g., save to disk, parse Excel)
        // You can use Apache POI or similar libraries to handle Excel files
        return "File uploaded successfully.";
    }
}

**Comment by anishcana** [comment]
// upload.js
$(document).ready(function() {
    $("#uploadForm").submit(function(event) {
        event.preventDefault();
        var formData = new FormData($(this)[0]);
        uploadFile(formData);
    });
});

function uploadFile(formData) {
    $.ajax({
        url: "/upload",
        type: "POST",
        data: formData,
        contentType: false,
        processData: false,
        success: function(response) {
            // Handle success
            console.log("File uploaded successfully:", response);
        },
        error: function(xhr, status, error) {
            // Handle error
            console.error("Error uploading file:", error);
        }
    });
}

**Comment by anishcana** [comment]
load Excel File</h1>
    <form id="uploadForm" enctype="multipart/form-data">
        <input type="file" name="file">
        <button type="submit">Upload</button>

**Comment by anishcana** [comment]
<h1>Upload Excel File</h1>
    <form id="uploadForm" enctype="multipart/form-data">
        <input type="file" name="file">
        <button type="submit">Upload</button>

**Comment by anishcana** [comment]
// upload.js
$(document).ready(function() {
    $("#uploadButton").click(function() {
        $("#fileInput").click();
    });

    $("#uploadForm").submit(function(event) {
        event.preventDefault();
        var formData = new FormData($(this)[0]);
        uploadFile(formData);
    });
});

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html>
<head>
    <title>Upload Excel File</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="upload.js"></script>
</head>
<body>
    <h1>Upload Excel File</h1>
    <button id="uploadButton">Choose File</button>
    <form id="uploadForm" enctype="multipart/form-data" style="display: none;">
        <input type="file" name="file" id="fileInput" style="display: none;">
        <button type="submit">Upload</button>
    </form>
</body>
</html>

**Comment by anishcana** [comment]
<dependency>
    <groupId>javax.validation</groupId>
    <artifactId>validation-api</artifactId>
</dependency>

**Comment by anishcana** [comment]
ependency>
    <groupId>javax.validation</groupId>
    <artifactId>validation-api</artifactId>
</dependenc

**Comment by anishcana** [comment]
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class YourController {

    @PostMapping("/createRecord")
    public ResponseEntity<String> createRecord(@RequestBody YourDTO formData) {
        // Code to save the formData to the database
        // You can use your service layer to handle this
        // Return appropriate response
        return ResponseEntity.ok("New record created successfully!");
    }
}

**Comment by anishcana** [comment]
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class YourController {

    @PostMapping("/editRecord")
    public ResponseEntity<String> editRecord(@RequestBody YourDTO formData) {
        // Code to update the existing record in the database
        // You can use your service layer to handle this
        // Return appropriate response
        return ResponseEntity.ok("Record edited successfully!");
    }
}

**Comment by anishcana** [comment]
<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Form</title>
</head>
<body>
    <h1>Create or Edit Record</h1>
    <form action="/createRecord" method="post">
        <label for="field1">Field 1:</label>
        <input type="text" id="field1" name="field1"><br>
        
        <label for="field2">Field 2:</label>
        <input type="text" id="field2" name="field2"><br>
        
        <label for="field3">Field 3:</label>
        <input type="text" id="field3" name="field3"><br>
        
        <button type="submit">Submit</button>
    </form>
</body>
</html>

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Dropdown Example</title>
</head>
<body>
    <h1>Dropdown Example</h1>
    <form action="/processForm" method="post">
        <label for="dropdown">Select an option:</label>
        <select id="dropdown" name="dropdown">
            <option value="option1">Option 1</option>
            <option value="option2">Option 2</option>
            <option value="option3">Option 3</option>
        </select>
        <button type="submit">Submit</button>
    </form>
</body>
</html>

**Comment by anishcana** [comment]
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.*;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

@RestController
public class ZipController {

    @Value("${source.file.path}")
    private String sourceFilePath;

    @Value("${destination.zip.path}")
    private String destinationZipPath;

    @PostMapping("/zip")
    public void zipFiles() throws IOException {
        FileOutputStream fos = new FileOutputStream(destinationZipPath);
        ZipOutputStream zos = new ZipOutputStream(fos);
        zipFile(sourceFilePath, sourceFilePath, zos);
        zos.close();
    }

    private void zipFile(String fileToZip, String fileName, ZipOutputStream zos) throws IOException {
        File file = new File(fileToZip);
        FileInputStream fis = new FileInputStream(file);
        ZipEntry zipEntry = new ZipEntry(fileName);
        zos.putNextEntry(zipEntry);
        byte[] bytes = new byte[1024];
        int length;
        while ((length = fis.read(bytes)) >= 0) {
            zos.write(bytes, 0, length);
        }
        zos.closeEntry();
        fis.close();
    }
}


source.file.path=/path/to/your/source/file
destination.zip.path=/path/to/your/destination/zipfile.zip

**Comment by anishcana** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Zip Files</title>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
</head>
<body>

<button id="zipButton">Zip Files</button>

<script>
    $(document).ready(function () {
        $("#zipButton").click(function () {
            $.ajax({
                url: "/zip",
                type: "POST",
                success: function () {
                    alert("Files zipped successfully!");
                },
                error: function () {
                    alert("Error zipping files!");
                }
            });
        });
    });
</script>

</body>
</html>
